plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceUtils.java

```diff
@@ -206,6 +207,10 @@ public static DBPDataSourceContainer getDataSourceBySpec(
                         Map<String, String> handlerPopMap = handlerProps.computeIfAbsent(handlerId, k -> new HashMap<>());
                         handlerPopMap.put(paramName, paramValue);
                         handled = true;
+                    } else if (paramName.length() > PREFIX_AVANCED_PROP.length() && paramName.startsWith(PREFIX_AVANCED_PROP)) {
```

Comment:
Redundant check; `startsWith` already checks the length
```suggestion
                    } else if (paramName.startsWith(PREFIX_AVANCED_PROP)) {
```